### PR TITLE
Add name to select control popover slots

### PR DESCRIPTION
**Pull request by louwie17**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds a name to the Popover and PopoverSlot so the `Popover.Slot` doesn't get used for all the popovers within the page.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Load this branch, build it, and make sure you have the `new-product-management-experience` feature flag enabled (you can do so using the latest version of the Beta tester within the mono repo).
2. Go to **Products > Add new (MVP)** and scroll down to the attributes section
3. Click **Add first attribute** and make sure the select control dropdown still shows correctly.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/components/changelog/fix-select-control-popover-slots [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Add name to exported popover slot so that it is not used for any popover items.
```
Comment (mattsherman) [inline packages/js/components/changelog/fix-select-control-popover-slots]: "not used for any popover items" should probably be "not used for all popover items"

Comment (mattsherman) [reply packages/js/components/changelog/fix-select-control-popover-slots]: Oh yeah... would probably be good to mention the context of what component we are talking about.

**Review by mattsherman — CHANGES_REQUESTED** [review]
Works well. Nice change!

As mentioned in Slack, I had a reservation about the use of `__unstableSlotName`, but it doesn't really look too unstable from what you mentioned:

- PR that added it was 1.5 years ago: https://github.com/WordPress/gutenberg/pull/31611
- There is a wishlist issue to remove the `unstable` part: https://github.com/WordPress/gutenberg/issues/42770

Only change I suggested was updating the changelog to be a bit more descriptive.

File: packages/js/components/changelog/fix-select-control-popover-slots (hunk 1 above)
Comment (louwie17) [reply packages/js/components/changelog/fix-select-control-popover-slots]: 🤣 yeah that is very ambiguous

File: packages/js/components/changelog/fix-select-control-popover-slots [hunk 2]
```diff
@@ -1,4 +1,4 @@
 Significance: patch
 Type: fix
 
-Add name to exported popover slot so that it is not used for any popover items.
+Add name to exported popover slot used to display SelectControl Menu, so it is only used for SelectControl menu's.
```
Comment (mattsherman) [inline packages/js/components/changelog/fix-select-control-popover-slots]: Ugh, I hate to be picky... but, it should be "menus", not "menu's"

**Review by mattsherman — CHANGES_REQUESTED** [review]

**Review by mattsherman — APPROVED** [review]
Thanks for making the changelog change. Looks good!

**Pull request merged**